modules/Bio/EnsEMBL/Compara/RunnableDB/HomologyAnnotation/DiamondBlastp.pm

```diff
@@ -0,0 +1,88 @@
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+Copyright [2016-2020] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 NAME
+
+Bio::EnsEMBL::Compara::RunnableDB::HomologyAnnotation::DiamondBlastp
+
+=head1 DESCRIPTION
+
+Create fasta file containing batch_size number of sequences. Run DIAMOND blastp and parse
+the output into PeptideAlignFeature objects. Store PeptideAlignFeature objects in the compara
+database
+
+=cut
+
+package Bio::EnsEMBL::Compara::RunnableDB::HomologyAnnotation::DiamondBlastp;
+
+use strict;
+use warnings;
+
+use base ('Bio::EnsEMBL::Compara::RunnableDB::BlastAndParsePAF');
+
+sub get_queries {
+    my $self = shift @_;
+
+    my $start_member_id = $self->param_required('start_member_id');
+    my $end_member_id   = $self->param_required('end_member_id');
+
+    #Get list of members and sequences
+    my $member_ids = $self->compara_dba->get_HMMAnnotAdaptor->fetch_all_seqs_missing_annot_by_range($start_member_id, $end_member_id, 'no_null');
+    return $self->compara_dba->get_SeqMemberAdaptor->fetch_all_by_dbID_list($member_ids);
+}
+
+sub run {
+    my $self = shift @_;
+
+    my $diamond_exe           = $self->param('diamond_exe');
+    my $blast_params          = $self->param('blast_params')  || '';  # no parameters to C++ binary means having composition stats on and -seg masking off
+    my $evalue_limit          = $self->param('evalue_limit');
+    my $worker_temp_directory = $self->worker_temp_directory;
+    my $blast_infile          = $worker_temp_directory . '/blast.in.' . $$;
+    my $blast_outfile         = $worker_temp_directory . '/blast.out.' . $$;
+
+    Bio::EnsEMBL::Compara::Utils::Preloader::load_all_sequences($self->compara_dba->get_SequenceAdaptor, undef, $self->param('query_set'));
+
+    my $members = $self->param('query_set')->get_all_Members;
```

Comment:
I can't see where `$members` is used in this file

Reply:
Meant to delete that, it was part of the debugging :+1: